packages/app/cypress/e2e/runner/reporter.errors.cy.ts

```diff
@@ -1,90 +1,865 @@
-import { verify } from './support/verify-helpers'
+import { createVerify, verifyInternalFailure } from './support/verify-failures'
+
+type LoadSpecOptions = {
+  fileName: string
+  onLoadStatsMessage: string
+  hasPreferredIde?: boolean
+}
+
+type VerifyFunc = (specTitle: string, verifyOptions: any) => void
+
+/**
+ * Navigates to desired spec file within Cypress app and waits for completion.
+ * Returns scoped verify function to aid inner spec validation.
+ */
+function loadSpec (options: LoadSpecOptions): VerifyFunc {
+  const {
+    fileName,
+    onLoadStatsMessage,
+    hasPreferredIde = false,
+  } = options
+
+  cy.scaffoldProject('runner-e2e-specs')
+  cy.openProject('runner-e2e-specs')
+  cy.startAppServer()
+
+  cy.withCtx((ctx, options) => {
+    if (options.hasPreferredIde) {
+      // set preferred editor to bypass IDE selection dialog
+      ctx.coreData.localSettings.availableEditors = [
+        ...ctx.coreData.localSettings.availableEditors,
+        {
+          id: 'test-editor',
+          binary: '/usr/bin/test-editor',
+          name: 'Test editor',
+        },
+      ]
+
+      ctx.coreData.localSettings.preferences.preferredEditorBinary = 'test-editor'
+    }
+
+    ctx.coreData.localSettings.preferences.isSpecsListOpen = false
+  }, { hasPreferredIde })
+
+  // directly visiting the spec will sometimes hang, going through
+  // specs page first to mitigate
+  // cy.visitApp(`specs/runner?file=cypress/e2e/errors/${fileName}`)
+
+  cy.visitApp()
+  cy.contains('[data-cy=spec-item]', fileName).click()
+
+  cy.location().should((location) => {
+    expect(location.hash).to.contain(fileName)
+  })
+
+  // Wait for specs to complete
+  cy.findByLabelText('Stats', { timeout: 10000 })
+  .get('.failed', { timeout: 10000 }).should('have.text', onLoadStatsMessage)
+
+  // Return scoped verify function with spec options baked in
+  return createVerify({ fileName, hasPreferredIde })
+}
 
 describe('errors ui', {
   viewportHeight: 768,
   viewportWidth: 1024,
+  // Limiting tests kept in memory due to large memory cost
+  // of nested spec snapshots
+  numTestsKeptInMemory: 1,
 }, () => {
-  describe('assertion failures', () => {
-    beforeEach(() => {
-      cy.scaffoldProject('runner-e2e-specs')
-      cy.openProject('runner-e2e-specs')
-
-      // set preferred editor to bypass IDE selection dialog
-      cy.withCtx((ctx) => {
-        ctx.coreData.localSettings.availableEditors = [
-          ...ctx.coreData.localSettings.availableEditors,
-          {
-            id: 'test-editor',
-            binary: '/usr/bin/test-editor',
-            name: 'Test editor',
-          },
-        ]
-
-        ctx.coreData.localSettings.preferences.preferredEditorBinary = 'test-editor'
-      })
+  it('assertion failures', () => {
```

Comment:
Why change these to multiple different tests inside of a single `it`? Previously, you could do `verify.it.only` and only run one test. Now you can only narrow it down to however many tests happen to be in an `it`.

Reply:
Before, each `verify.it` would load the spec file and verify the output for a single spec within it. However, many of these spec files contained more than one spec, so there was a lot of redundant execution. 

With the conversion to a more typical e2e test, it seemed like overkill to reload and wait for the entire suite to finish to assert on a single spec within it. So this instead loads the suite once, waits for it to finish once, and then performs all the assertions for each spec in that suite (in a single `it`). With this we go from 85 suite loads to about 25, saving us quite a bit of test time.

I agree that the granularity of having individual `it`s and the ability to use `.skip`/`.only` was convenient.  I had attempted to get the best of both worlds by performing a single load `before` the individual `it`s, but my understanding is that will not be possible with upcoming session changes. Another option would have been to break up the specs so that there is less execution redundancy, but that would have left us with a 85 individual spec files to maintain (and the overhead for loading/running each one).

I think the current implementation is a fair balance between maintaining existing coverage and overall test performance, but any input is good to have on this strategy before more tests get migrated.

Reply:
The previous method [only runs a single test in the inner layer spec](https://github.com/cypress-io/cypress/blob/589f93b11f1c5dc74cd4ffed06c847887756fe82/packages/runner/cypress/fixtures/errors/setup.js#L1-L8), the one the outer layer is verifying. It probably still takes more time to execute than running multiple at once, but I'd be curious what that time difference is.

Reply:
The current reporter.errors.spec.js in develop runs pretty quick, just over 2 minutes in open mode. This PR as written runs in right around 3. Updating this PR to use individual specs again results in a _significantly_ longer runtime at 8+ minutes, which is what I was trying to avoid.

It's hard to get an apples to apples comparison between the two, as there are many differences between the code bases at this point, but I imagine the existing spec is getting a decent boost from all the mocking done with `runIsolatedCypress` vs. a the hands-off approach with the cypress-in-cypress testing setup.

But I think at the end of the day that's the question though. Should we prioritize getting these runner tests that use `runIsolatedCypress` migrated to app but otherwise unchanged, or should we convert them to more representative e2e tests, even if it means they change somewhat? I think there's value in moving towards consistency with the rest of the e2e tests, but I definitely don't want to loose coverage or inflate the total execution time too much.

Reply:
If it's that significant a time difference, I'm okay with leaving it the way you have it.